### PR TITLE
upgrading to sbt 1

### PR DESCRIPTION
**Pull request by yasuba**
This is upgrading sbt, scala version and various dependencies. 

A lot of confusion around this: `The evaluation of `streams` inside an anonymous function is prohibited.` and ` `value` can only be called on a task within a task definition macro, such as :=, +=, ++=, or Def.task.` led me to create lots of `logStream` functions which is not cool. I'd appreciate some suggestions on how to do this properly!

### CODE REVIEW

File: sbt-chuckwagon/src/main/scala/com/itv/chuckwagon/sbt/ChuckwagonCopyPlugin.scala [hunk 1]
```diff
@@ -37,6 +37,18 @@ object ChuckwagonCopyPlugin extends AutoPlugin {
 
         val maybeVpcConfig = maybeVpcConfigFromProductionLambdaConfiguration().value
 
+        maybeVpcConfig.foreach { vpcConfig =>
+          streams.value.log.info(
```
Comment (agustafson) [inline sbt-chuckwagon/src/main/scala/com/itv/chuckwagon/sbt/ChuckwagonCopyPlugin.scala]: can use `logStream` here?

**Review by agustafson — APPROVED** [review]
LGTM 👍 
Thanks for updating!

**Comment by agustafson** [comment]
Don't really know how you have a single def of the `logStream`

File: build.sbt [hunk 2]
```diff
@@ -11,8 +11,8 @@ lazy val commonSettings = Seq(
 sonatypeProfileName := "com.itv"
 
 lazy val noPublishSettings = Seq(
-  publish := (),
-  publishLocal := (),
+  publish := Unit,
+  publishLocal := Unit,
```
Comment (davesmith00000) [inline build.sbt]: Surely not?

File: aws-scala-sdk/src/main/scala/com/itv/aws/package.scala [hunk 3]
```diff
@@ -11,11 +11,10 @@ package object aws {
...
       builder: AwsSyncClientBuilder[Builder, Client]
-  ): Regions => AWSCredentialsProvider => Client = {
-    (region: Regions) => credentials: AWSCredentialsProvider =>
-      {
-        builder.withRegion(region).withCredentials(credentials).build
-      }
+  ): Regions => AWSCredentialsProvider => Client = { (region: Regions) => credentials: AWSCredentialsProvider =>
+    {
+      builder.withRegion(region).withCredentials(credentials).build
+    }
```
Comment (davesmith00000) [inline aws-scala-sdk/src/main/scala/com/itv/aws/package.scala:17]: As a completely separate commit / PR - I think you should run scalafmt over the whole code base.

File: build.sbt [hunk 4]
```diff
@@ -40,11 +40,9 @@ lazy val publishSettings = Seq(
 val awsSdkVersion = "1.11.119"
 val circeVersion  = "0.7.1"
 val slf4jVersion  = "1.7.25"
-val scala212      = "2.12.2"
-val scala210      = "2.10.6"
 
 lazy val root = (project in file("."))
-  .enablePlugins(CrossPerProjectPlugin)
+  .enablePlugins()
```
Comment (davesmith00000) [inline build.sbt]: And so this is redundant?

**Review by davesmith00000 — COMMENTED** [review]
Couple of minor comments but it looks plausible.

File: build.sbt (hunk 2 above)
Comment (yasuba) [reply build.sbt]: The compiler told me to do it!

Comment (yasuba) [reply build.sbt]: Well, it said that () was wrong, at least.

File: build.sbt (hunk 4 above)
Comment (yasuba) [reply build.sbt]: Good point. I was so busy removing the things that made the compiler unhappy that I didn't think about whether what was left was needed.

**Pull request merged**